src/core/handlers/RequestHandler.ts

```diff
@@ -263,10 +263,20 @@ export abstract class RequestHandler<
       request: args.request,
       parsedResult,
     })
-    const mockedResponse = (await executeResolver({
-      ...resolverExtras,
-      request: args.request,
-    })) as Response
+
+    const mockedResponsePromise = (
+      executeResolver({
+        ...resolverExtras,
+        request: args.request,
+      }) as Promise<Response>
+    ).catch((errorOrResponse) => {
```

Comment:
I'm adding this on the lowest level possible for two reasons:

1. This way all handlers get this behavior, no matter if the `handleRequest` is run in the browser or Node context;
2. This way even the direct users of the `RequestHandler` get this behavior, which is essential.